### PR TITLE
ssh proxy (added logic for ssh tunnel closing)

### PR DESCRIPTION
**Pull request by epankou**
changes for ssh proxy

### CODE REVIEW

File: daemon/src/org/eclipse/iofog/field_agent/FieldAgent.java [hunk 1]
```diff
@@ -772,6 +772,10 @@ public void postFogConfig() throws Exception {
 		}
 	}
 
+	/**
+	 * gets IOFog proxy configuration from IOFog controller
+	 * @throws Exception exception
+	 */
 	public void getProxyConfig() throws Exception {
```
Comment (elukashick) [inline daemon/src/org/eclipse/iofog/field_agent/FieldAgent.java]: What the point of 'throws Exception' here?

Comment (epankou) [reply daemon/src/org/eclipse/iofog/field_agent/FieldAgent.java]: already written method `controllerNotConnected` throws Exception.
I just had to reuse that code

Comment (epankou) [reply daemon/src/org/eclipse/iofog/field_agent/FieldAgent.java]: fixed

File: daemon/src/org/eclipse/iofog/proxy/SshProxyManager.java [hunk 2]
```diff
@@ -0,0 +1,179 @@
...
+            handleAlreadyOpenedTunnel();
+        } else {
+            handleAlreadyClosedTunnel();
+        }
+    }
+
+    /**
+     * reports that the tunnel is already opened
+     */
+    private void handleAlreadyOpenedTunnel() {
```
Comment (epankou) [inline daemon/src/org/eclipse/iofog/proxy/SshProxyManager.java]: make only one method

**Pull request merged**